### PR TITLE
`ActiveSupport::TimeWithZone` inherits `Time`

### PR DESCRIPTION
**Pull request by tonybruess**
According to documentation, `ActiveSupport::TimeWithZone` implements all `Time` methods. Is this the best way to do this?

https://api.rubyonrails.org/classes/ActiveSupport/TimeWithZone.html

### CODE REVIEW

File: rbi/annotations/activesupport.rbi [hunk 1]
```diff
@@ -46,3 +46,6 @@ class Object
   sig { returns(T::Boolean) }
   def present?; end
 end
+
+class ActiveSupport::TimeWithZone < Time
+end
```
Comment (KaanOzkan) [inline rbi/annotations/activesupport.rbi:51]: I think representing it as a mixin is slightly more accurate and you can satisfy our CI using `@shim`, similar to https://github.com/Shopify/rbi-central/blob/cd36dd5bc194097e703c384d8e7b94370be45df8/rbi/annotations/activerecord.rbi#L9

**Review by KaanOzkan — COMMENTED** [review]
Missing methods are being delegated to `Time` so this looks good to me 

https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/activesupport/lib/active_support/time_with_zone.rb#L567-L573

File: rbi/annotations/activesupport.rbi (hunk 1 above)
Comment (tonybruess) [reply rbi/annotations/activesupport.rbi:51]: Perf, will change

**Comment by tonybruess** [comment]
Change to a mixin, fixed the build ◡̈

**Comment by KaanOzkan** [comment]
@tonybruess let's identify the methods missing between `Time` and `TimeWithZone` and manually define them please. I was under incorrect assumption that inheritance would be viable but we don't want exactly what it entails. I've added an explanation on the tapioca PR too https://github.com/Shopify/tapioca/pull/1352#issuecomment-1379548571. Sorry about the incorrect back and forth ❤️

**Pull request closed without merging**